enterprise/api_client/enterprise.py

```diff
@@ -30,19 +30,20 @@ class EnterpriseApiClient(JwtLmsApiClient):
 
     DEFAULT_VALUE_SAFEGUARD = object()
 
-    def get_content_metadata(self, enterprise_customer):
+    def get_content_metadata(self, enterprise_customer, catalogs_to_transmit=None):
         """
         Return all content metadata contained in the catalogs associated with the EnterpriseCustomer.
 
         Arguments:
             enterprise_customer (EnterpriseCustomer): The EnterpriseCustomer to return content metadata for.
+            catalogs_to_transmit (EnterpriseCustomerCatalog): Optional list of EnterpriseCustomerCatalog objects
 
         Returns:
             list: List of dicts containing content metadata.
         """
         content_metadata = OrderedDict()
-
-        for enterprise_customer_catalog in enterprise_customer.enterprise_customer_catalogs.all():
+        enterprise_customer_catalogs = catalogs_to_transmit or enterprise_customer.enterprise_customer_catalogs.all()
```

Comment:
we should pull catalogs which needs to be transmitted from enterprise->enterprise channel config->catalogs relationship instead of passing it in `catalogs_to_transmit` param.